debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlTaskContext.java

```diff
@@ -246,4 +247,41 @@ public ObjectName metricName(String contextName) throws MalformedObjectNameExcep
         return new ObjectName("debezium.mysql:type=connector-metrics,context=" + contextName + ",server=" + serverName());
     }
 
+    /**
+     * Retrieve GTID set after applying include/exclude filters on the source. Also merges the server GTID set with
+     * the filtered set. The merging is required when MySQL has multiple channels, and Debezium is only consuming a data from a
+     * subset of those channels.
+     *
```

Comment:
Probably should say that this does not affect any state within the context. Also, would be good to explain what "merging" actually means, and why we even need to do this (and perhaps why doing this works).